No implementations provided error persists with opam-file-format 2.1.2

Still getting the same error as with opam-parser-types 2.1.1 when compiling opaline:

```
ocamlbuild -use-ocamlfind opaline.native
/bin/sh: tput: not found
+ ocamlfind ocamlc -c -package unix,opam-file-format -o opaline.cmo opaline.mlK
File "opaline.ml", line 177, characters 20-35:
177 |     let opam_file = OpamParser.file f in
                          ^^^^^^^^^^^^^^^
Alert deprecated: OpamParser.file
Use OpamParser.FullPos.file instead.
+ ocamlfind ocamlopt -c -package unix,opam-file-format -o opaline.cmx opaline.ml
File "opaline.ml", line 177, characters 20-35:
177 |     let opam_file = OpamParser.file f in
                          ^^^^^^^^^^^^^^^
Alert deprecated: OpamParser.file
Use OpamParser.FullPos.file instead.
+ ocamlfind ocamlopt -linkpkg -package unix,opam-file-format opaline.cmx -o opaline.native
File "_none_", line 1:
Error: No implementations provided for the following modules:
         OpamParserTypes referenced from /nix/store/m81fjglbljz64ang0q29hsfpbmdbc9pk-ocaml4.10.0-opam-file-format-2.1.2/lib/ocaml/4.10.0/site-lib/opam-file-format/opam-file-format.cmxa(OpamParser)
Command exited with code 2.
Compilation unsuccessful after building 4 targets (0 cached) in 00:00:00.
make: *** [Makefile:13: opaline.native] Error 10
```

`OpamParserTypes` seems to be compiled properly, now generating an object file:

```
> tree result/
result/
└── lib
    └── ocaml
        └── 4.10.0
            └── site-lib
                └── opam-file-format
                    ├── META
                    ├── opamBaseParser.cmi
                    ├── opamBaseParser.cmo
                    ├── opamBaseParser.cmx
                    ├── opam-file-format.a
                    ├── opam-file-format.cma
                    ├── opam-file-format.cmxa
                    ├── opamLexer.cmi
                    ├── opamLexer.cmo
                    ├── opamLexer.cmx
                    ├── opamParser.cmi
                    ├── opamParser.cmo
                    ├── opamParser.cmx
                    ├── opamParserTypes.cmi
                    ├── opamParserTypes.cmo
                    ├── opamParserTypes.cmx
                    ├── opamPrinter.cmi
                    ├── opamPrinter.cmo
                    └── opamPrinter.cmx
```

See #36 for context.

--- a/CHANGES
+++ b/CHANGES
@@ -1,3 +1,6 @@
+2.1.3 [???]
+* Fix module link order (broken by #37) [#41 @dra27]
+
 2.1.2 [07 Jan 2021]
 * Some hash-consing for strings [#27 @AltGr]
 * Fix named section position [#31 @rjbou]

--- a/dune-project
+++ b/dune-project
@@ -1,1 +1,1 @@
-(lang dune 1.0)
+(lang dune 1.3)

--- a/opam-file-format.opam
+++ b/opam-file-format.opam
@@ -1,5 +1,5 @@
 opam-version: "2.0"
-version: "2.1.2"
+version: "2.1.3"
 synopsis: "Parser and printer for the opam file syntax"
 maintainer: "Louis Gesbert <louis.gesbert@ocamlpro.com>"
 authors: "Louis Gesbert <louis.gesbert@ocamlpro.com>"

--- a/opam-file-format.opam
+++ b/opam-file-format.opam
@@ -20,3 +20,4 @@ depends: [
 depopts: [
   "dune"
 ]
+conflicts: "dune" {< "1.3.0"}

--- a/src/Makefile
+++ b/src/Makefile
@@ -31,20 +31,20 @@ opamParserTypes.cmi: opamParserTypes.cmo
 %.cmx: %.ml
 	ocamlopt -c $(PP) $<
 
-MODULES = $(sort $(basename $(wildcard *.ml *.mll *.mly)))
+MODULES = opamParserTypes opamBaseParser opamLexer opamParser opamPrinter
 
 GENERATED = $(patsubst %.mly,%.ml,$(wildcard *.mly)) \
             $(patsubst %.mly,%.mli,$(wildcard *.mly)) \
             $(patsubst %.mll,%.ml,$(wildcard *.mll)) \
 	    META
 
-opam-file-format.cma: $(MODULES:%=%.cmo)
+opam-file-format.cma: $(addsuffix .cmo,$(MODULES))
 	ocamlc -a $^ -o $@
 
-opam-file-format.cmxa: $(MODULES:%=%.cmx)
+opam-file-format.cmxa: $(addsuffix .cmx,$(MODULES))
 	ocamlopt -a $^ -o $@
 
-opam-file-format.cmxs: $(MODULES:%=%.cmx)
+opam-file-format.cmxs: $(addsuffix .cmx,$(MODULES))
 	ocamlopt -shared $^ -o $@
 
 META: META.in ../opam-file-format.opam $(wildcard *.cm*)

--- a/tests/legacy/Makefile
+++ b/tests/legacy/Makefile
@@ -1,0 +1,11 @@
+.PHONY: test
+test: legacy.byte legacy.native
+
+src/%:
+	$(MAKE) -C src $*
+
+legacy.byte: src/opam-file-format.cma legacy.ml
+	ocamlc -o $@ -I src $^
+
+legacy.native: src/opam-file-format.cmxa legacy.ml
+	ocamlopt -o $@ -I src $^

--- a/tests/legacy/dune
+++ b/tests/legacy/dune
@@ -1,0 +1,8 @@
+; This test (driven by Dune) ensures that the old build system works
+
+(copy_files ../../opam-file-format.opam)
+
+(alias
+  (name runtest)
+  (deps Makefile legacy.ml opam-file-format.opam (glob_files src/*))
+  (action (run make test)))

--- a/tests/legacy/legacy.ml
+++ b/tests/legacy/legacy.ml
@@ -1,0 +1,3 @@
+(* This "program" is used to verify that the library is linkable - see #40 *)
+let file = OpamParser.FullPos.file "../opam-file-format.opam"
+in Printf.printf "Successfully loaded %s\n" file.OpamParserTypes.FullPos.file_name

--- a/tests/legacy/src/dune
+++ b/tests/legacy/src/dune
@@ -1,0 +1,3 @@
+; We mustn't copy the dune file, but Dune's globbing has no syntax for that
+(copy_files ../../../src/o*)
+(copy_files ../../../src/M*)

